common/strings/patch.cc

```diff
@@ -146,8 +149,10 @@ absl::Status HunkHeader::Parse(absl::string_view text) {
 }
 
 std::ostream& operator<<(std::ostream& stream, const HunkHeader& header) {
-  return stream << "@@ -" << header.old_range << " +" << header.new_range
-                << " @@" << header.context;
+  return stream << term::StartColor(stream, term::Color::kGreen) << "@@ -"
+                << header.old_range << " +" << header.new_range << " @@"
+                << header.context
+                << term::EndColor(stream, term::Color::kGreen);
```

Comment:
Can we provide an interface that uses RAII to automatically change back to the previous color setting, so no one has to ever write `EndColor`?

Reply:
What you suggest was my initial approach, but this snippet of code made me change my mind as I didn't want to change many things.

The other alternative I see is doing something like this:

```c++

return stream << term::Color(stream, kGreen, StrCat("@@ -", header.old_range, ...));
```

I don't have a clear preference, so happy to adapt it either way you find cleaner

Reply:
Sorry, I missremembered this earlier. The reason I went with Start/EndColor is because in this function we rely on the overloaded operator<< for HunkIndices, so I didn't want to mess with that. 

The easiest fix I see is removing operator<< for HunkIndices and substituting it for a to_string() method so we could do the following

```c++
std::ostream& operator<<(std::ostream& stream, const HunkHeader& header) {
  return term::Colored(
      stream,
      absl::StrCat("@@ -", header.old_range.to_string(), " +",
                   header.new_range.to_string(), " @@", header.context),
      term::Color::kGreen);
}
```